File: BPG_CMS_BE/src/BPG.Domain/Entities/PhaseAcceptance.cs
namespace BPG.Domain.Entities;

public class PhaseAcceptance : BaseEntity
{
    public long AcceptanceId { get; set; }
    public long PhaseId { get; set; }
    public long AcceptedBy { get; set; }
    public DateTime AcceptanceDate { get; set; }
    public string ReportContent { get; set; } = string.Empty;
    public string? PdfUrl { get; set; }
    public bool IsCancelled { get; set; } = false;
    public string? CancellationReason { get; set; }
    public DateTime? CancelledAt { get; set; }
    public long? CancelledBy { get; set; }

    public Phase Phase { get; set; } = null!;
    public User Acceptor { get; set; } = null!;
}

File: BPG_CMS_FE/src/types/phaseAcceptance.ts


export interface PhaseAcceptance {
  acceptanceId: number;
  phaseId: number;
  phaseName: string;
  projectName: string;
  acceptedBy: number;
  acceptedByName: string;
  acceptanceDate: string;
  reportContent: string;
  pdfUrl?: string;
  isCancelled: boolean;
  cancellationReason?: string;
  cancelledAt?: string;
  cancelledByName?: string;
}

export interface GetPhaseAcceptancesQuery {
  pageIndex: number;
  pageSize: number;
  projectId?: number;
  phaseId?: number;
}

export interface AcceptPhaseCommand {
  phaseId: number;
  reportContent: string;
}

export interface CancelAcceptanceRequest {
  cancellationReason: string;
}
